src/App.tsx

```diff
@@ -1,14 +1,50 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { PrepaparedTodo } from './types/PreparedTodo';
+import { FilterBy } from './types/FilterBy';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [filterBy, setFilterBy] = useState(FilterBy.ALL);
+  const [selectedTodo, setSelectedTodo] = useState<PrepaparedTodo | null>(null);
+  const [isModalOpen, setIsOpenModal] = useState(false);
+
+  const handleCloseModal = () => {
+    setSelectedTodo(null);
+    setIsOpenModal(false);
+  };
+
+  useEffect(() => {
+    getTodos()
+      .then(todosFromServer => {
+        setTodos(todosFromServer);
+      });
+  }, []);
+
+  const filteredTodos = useMemo(() => (
+    todos.filter(todo => {
+      const filtered = todo.title.toLowerCase().includes(query.toLowerCase().trim());
+
+      switch (filterBy) {
+        case FilterBy.ACTIVE:
+          return filtered && !todo.completed;
+        case FilterBy.COMPLETED:
+          return filtered && todo.completed;
```

Comment:
```suggestion
          return !todo.completed;
        case FilterBy.COMPLETED:
          return todo.completed;
```

Reply:
But if I remove this 'filtered', the search will not be performed when active or completed is selected